src/components/BoringHamburger.vue

```diff
@@ -0,0 +1,36 @@
+<template>
+  <plain-hamburger class="hamburger--boring" @toggle="$emit('toggle')" />
```

Comment:
I would automatically pass any event to the parent with `v-on="$listeners"`
```suggestion
  <plain-hamburger class="hamburger--boring" v-on="$listeners />
```

PS. the same for the other components